src/backends/odbc/vector-use-type.cpp

```diff
@@ -204,31 +224,60 @@ void* odbc_vector_use_type_backend::prepare_for_bind(SQLUINTEGER &size,
             std::size_t maxSize = 0;
             std::size_t const vecSize = get_vector_size(type_, data_);
             prepare_indicators(vecSize);
-            for (std::size_t i = 0; i != vecSize; ++i)
+            for (std::size_t i = 0; i < vecSize; ++i)
             {
                 std::size_t sz = vector_string_value(type_, data_, i).length();
+
+#ifdef SOCI_ODBC_WIDE
+                set_sqllen_from_vector_at(i, static_cast<long>(sz * sizeof(SQLWCHAR)));
+#else
                 set_sqllen_from_vector_at(i, static_cast<long>(sz));
+#endif // SOCI_ODBC_WIDE
```

Comment:
Again, why not just

```suggestion
                set_sqllen_from_vector_at(i, static_cast<long>(sz * sizeof(SQLTCHAR)));
```

?

Reply:
At first I didn't know of the existence of SQLTCHAR and created my own alias. Because it can get confusing, I later removed that alias and did it this way. But I'm fine either way. I will make the necessary changes everywhere.